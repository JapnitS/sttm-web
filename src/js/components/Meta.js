import React, { createRef } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { withRouter } from 'react-router-dom';
import DatePicker from 'react-date-picker';
import AudioPlayer from 'react-h5-audio-player';

import CalendarIcon from './Icons/CalendarIcon';
import Chevron from './Icons/Chevron';
import Hour24 from './Icons/Hour24';
import ForwardIcon from './Icons/ForwardIcon';
import HeadphonesIcon from './Icons/HeadphonesIcon'
import TimesIcon from './Icons/Times';

import {
  isFalsy,
  toAngURL,
  toNavURL,
  dateMath,
  getSourceId,
  getWriter,
  getRaag,
  checkAPI,
  getShabadAudioUrl
} from '@/util';
import { TEXTS, PAGE_NAME, FIRST_HUKAMNAMA_DATE, HUKAMNAMA_AUDIO_URL, S3_BUCKET_URL, API_URL  } from '@/constants';

/**
 *
 *
 * @export
 * @class Meta
 * @augments {React.PureComponent<MetaProps>}
 */
class Meta extends React.PureComponent {

  constructor(props) {
    super(props)
    this.state = {
      audioPlayer: null,
      isHukamnamaAudioPlayerVisible: true,
      shabadURL: '',
      isShabadPlayable: false,
    }
    this.audioPlayerRef = createRef();
    this.audioPlayerIconRef = createRef();
    this.wrapperRef = createRef();
  }
  static defaultProps = {
    nav: {},
  };

  /**
   * @typedef {object} MetaProps
   * @property {ShabadContentTypes} type
   * @property {object} info
   * @property {array} translationLanguages
   * @property {array} transliterationLanguages
   * @property {object} nav
   * @property {boolean} isUnicode
   * @property {boolean} isArrowsHidden
   *
   * @static
   * @memberof Meta
   */
  static propTypes = {
    history: PropTypes.object.isRequired,
    type: PropTypes.string.isRequired,
    info: PropTypes.object.isRequired,
    translationLanguages: PropTypes.array.isRequired,
    transliterationLanguages: PropTypes.array.isRequired,
    isUnicode: PropTypes.bool.isRequired,
    nav: PropTypes.shape({
      current: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
      previous: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
      next: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    }),
    isArrowsHidden: PropTypes.bool,
    showPinSettings: PropTypes.bool,
  };

  renderLeftArrow() {
    const {
      type,
      nav = {},
      isArrowsHidden,
    } = this.props;

    const link = toNavURL(this.props);
    const isShowPreviousArrow = isFalsy(nav.previous) === false;
    const isSync = type === 'sync';
    const isHukamnama = type === 'hukamnama';

    if (!isArrowsHidden) {
      if (isShowPreviousArrow) {
        return (
          <div className="shabad-nav left">
            <Link to={link + nav.previous}>
              {isHukamnama ? (
                <div className='hukamnama-nav-icon'>
                  <Hour24 direction='previous' />
                  <span>{dateMath.expand(nav.previous, false)}</span>
                </div>
              ) : (
                <Chevron direction={Chevron.DIRECTIONS.LEFT} />
              )}
            </Link>
          </div>
        )
      }
      if (!isSync) {
        return (
          <div className="shabad-nav left disabled-nav">
            <a>
              {isHukamnama ? (
                <Hour24 direction='previous' />
              ) : (
                <Chevron direction={Chevron.DIRECTIONS.LEFT} />
              )}
            </a>
          </div>
        )
      }
    }

    return '';
  }


  renderRightArrow() {
    const {
      type,
      nav = {},
      isArrowsHidden,
    } = this.props;

    const isShowNextArrow = isFalsy(nav.next) === false
    const isSync = type === 'sync';
    const isHukamnama = type === 'hukamnama';

    if (!isArrowsHidden) {

      if (isShowNextArrow) {
        return (
          <div className="shabad-nav right">
            <a role="button" aria-label="next" onClick={this.goToNextAng}>
              {isHukamnama ? (
                <div className='hukamnama-nav-icon'>
                  <Hour24 direction='next' />
                  <span>{dateMath.expand(nav.next, false)}</span>
                </div>
              ) : (
                <Chevron direction={Chevron.DIRECTIONS.RIGHT} />
              )}
            </a>
          </div>
        )
      }

      if (!isSync) {
        return (
          <div className="shabad-nav right disabled-nav">
            <a>
              {isHukamnama ? (
                <Hour24 direction='next' />
              ) : (
                <Chevron direction={Chevron.DIRECTIONS.RIGHT} />
              )}
            </a>
          </div>
        )
      }
    }

    return '';
  }

  setShabadURL = (url) => {
    if (url === '') {
      this.setState(previousState => {
        return ({
          ...previousState,
          shabadURL: url
        })
      })
    }
  }

  setHukamnamaAudioPlayerVisibility = (e) => {
    e.preventDefault();
    const audioPlayer = this.audioPlayerRef.current.audio.current;

    this.setState(previousState => {
      // console.log('set state 2 times got hit')
      const isAudioPlayerState = !previousState.isHukamnamaAudioPlayerVisible;

      !isAudioPlayerState && audioPlayer.pause();

      return ({
        ...previousState,
        isHukamnamaAudioPlayerVisible: !previousState.isHukamnamaAudioPlayerVisible
      })
    })
  }

  removeAudioPlayer = (e) => {
    return this.setHukamnamaAudioPlayerVisibility(e);
  }

  async componentDidMount() {
    if (this.props.type === 'shabad' ) {
      const healthy = await checkAPI()
      if (healthy) {
        const audioUrl = await getShabadAudioUrl(this.props.info);
        this.setState(previousState => {
          return ({
            ...previousState,
            shabadURL: audioUrl,
            isShabadPlayable: (audioUrl!='')
          })
        });
      }
    }
  }
  
  render() {
    const {
      type,
      info,
      nav,
      isUnicode,
      translationLanguages,
      transliterationLanguages,
      showPinSettings,
    } = this.props;
    const Item = ({ children, last = false }) =>
      children ? (
        <React.Fragment>
          {children}
          {last ? '' : ' - '}
        </React.Fragment>
      ) : null;

    const shouldShowEnglishInHeader =
      translationLanguages.includes('english') ||
      transliterationLanguages.includes('english');
    const contentType = isUnicode ? 'unicode' : 'gurmukhi';
    const isHukamnama = type === 'hukamnama';
    const todayDate = new Date(new Date().toDateString());
    const hukamnamaDate = new Date(nav.current);
    const maximumHukamnamaDate = new Date(todayDate);
    const hasAudioPlayer = isHukamnama;
    // hukamnamaDate.getTime() == todayDate.getTime();

    return (
      <div ref={this.wrapperRef} id="metadata" className={`metadata-${type}`}>
        {!isHukamnama && this.renderLeftArrow()}

        <div className="meta">
          {isHukamnama && (
            <>
              <div className="meta-hukamnama">
                <div className="meta-hukamnama-left">
                  <div className="meta-hukamnama-left-row">
                    <DatePicker
                      isOpen={this.state.isCalendarOpen}
                      clearIcon={null}
                      onCalendarClose={() => {
                        this.setState(() => ({ isCalendarOpen: false }))
                      }}
                      onCalendarOpen={() => {
                        this.setState(() => ({ isCalendarOpen: true }))
                      }}
                      onChange={this.goToParticularHukamnama}
                      value={hukamnamaDate}
                      maxDate={maximumHukamnamaDate}
                      minDate={new Date(FIRST_HUKAMNAMA_DATE)}
                      calendarIcon={<CalendarIcon width={20} />}
                      calendarAriaLabel="date picker"
                    />
                    <a className="hukam-text-link" onClick={this.state.isCalendarOpen ? undefined : (e) => {
                      e.preventDefault();
                      return this.setState(() => ({ isCalendarOpen: true }))
                    }}>
                      <span title="Past Hukamnamas">Archive</span>
                    </a>
                  </div>
                  <div className="meta-hukamnama-left-gotoshabad">

                    <Link className="hukamnama-right-link" to={`/shabad?id=${info.shabadId}`}>
                      <ForwardIcon />{TEXTS.GO_TO_SHABAD}
                    </Link>
                  </div>
                </div>
                <h4>
                  {TEXTS.HUKAMNAMA_HEADING}, <span>{nav.current}</span>
                </h4>
                <div ref={this.audioPlayerIconRef} role='button' className="meta-hukamnama-right" onClick={this.setHukamnamaAudioPlayerVisibility}>
                  <span className="hukamnama-right-headphonesIcon"><HeadphonesIcon /><a title="Listen to Today's Hukamnama">{`Today's Hukamnama`}</a></span>
                </div>
              </div>
            </>
          )}
          <h4 className="gurbani-font">
            <Item>
              {getRaag(info) &&
                getRaag(info)[contentType] &&
                getRaag(info)[contentType] !== 'null' &&
                getRaag(info)[contentType]}
            </Item>
            <Item>{getWriter(info) && getWriter(info)[contentType]}</Item>
            <Item>{info.source[contentType]}</Item>
            {info.source.pageNo !== null && (
              <Item last>
                <Link
                  to={toAngURL({
                    ang: info.source.pageNo,
                    source: getSourceId(info),
                  })}
                >
                  {getSourceId(info) == 'G' ? PAGE_NAME.ANG[contentType] : PAGE_NAME.PANNA[contentType]}{' '}
                  {info.source.pageNo}
                </Link>
              </Item>
            )}
          </h4>

          {shouldShowEnglishInHeader && (
            <h4>
              <Item>
                {getRaag(info) &&
                  getRaag(info)['english'] &&
                  getRaag(info)['english'] !== 'null' &&
                  getRaag(info)['english']}
              </Item>
              <Item>{getWriter(info) && getWriter(info)['english']}</Item>
              <Item>{info.source.english}</Item>
              {info.source.pageNo !== null && (
                <Item last>
                  <Link
                    to={toAngURL({
                      ang: info.source.pageNo,
                      source: getSourceId(info),
                    })}
                  >
                    {getSourceId(info) == 'G' ? 'Ang' : 'Pannaa'}{' '}
                    {info.source.pageNo}
                  </Link>
                </Item>
              )}
            </h4>
          )}
          {this.state.isShabadPlayable && (
            <div ref={this.audioPlayerIconRef} role='button' className="meta-hukamnama-right" onClick={this.setHukamnamaAudioPlayerVisibility}>
              <span className="hukamnama-right-headphonesIcon"><HeadphonesIcon /><a title="Listen to this Shabad">{`Listen shabad`}</a></span>
            </div>
          )}
        </div>

        {hasAudioPlayer && (
          <div className={`hukamnama-audio ${this.state.isHukamnamaAudioPlayerVisible ? 'hukamnama-audio--shown' : 'hukamnama-audio--hidden'} ${showPinSettings ? 'hukamnama-audio--pin-settings' : ''}`}>
            <AudioPlayer
              ref={this.audioPlayerRef}
              src={HUKAMNAMA_AUDIO_URL}
              customAdditionalControls={[]}
              customVolumeControls={[]}
              header={(
                <div>
                  <h3 className="hukamnama-player-title">{`Listen to Today's Hukamnama`}</h3>
                  <span className="hukamnama-player-close-icon">
                    <TimesIcon onClick={this.removeAudioPlayer} />
                  </span>
                </div>
              )}
            />
          </div>)}
        
        {this.state.isShabadPlayable && (
          <div className={`hukamnama-audio ${(this.state.isHukamnamaAudioPlayerVisible && this.state.isShabadPlayable) ? 'hukamnama-audio--shown' : 'hukamnama-audio--hidden'} ${showPinSettings ? 'hukamnama-audio--pin-settings' : ''}`}>
          <AudioPlayer
              ref={this.audioPlayerRef}
              src={this.state.shabadURL}
              customAdditionalControls={[]}
              customVolumeControls={[]}
              header={(
                <div>
                  <h3 className="hukamnama-player-title">{`Listen to this Shabad`}</h3>
                  <span className="hukamnama-player-close-icon">
                    <TimesIcon onClick={this.removeAudioPlayer} />
                  </span>
                </div>
              )}
            />
          </div>
        )}
        {!isHukamnama && this.renderRightArrow()}
      </div >
    );
  }

  /**
   * Handle SaveAng
   * @memberof Meta
   */

  goToParticularHukamnama = (date) => {
    const month = date.getMonth() + 1;
    const year = date.getFullYear();
    const day = date.getDate();

    const hukamnamaDate = `${year}/${month}/${day}`;
    const link = toNavURL(this.props)
    this.props.history.push(link + hukamnamaDate);
  }
  goToNextAng = () => {
    const link = toNavURL(this.props);
    this.props.history.push(link + this.props.nav.next);
  };
}

export default withRouter(Meta);
