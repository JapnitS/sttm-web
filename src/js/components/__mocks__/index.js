export * from './shabad';
export * from './multipleShabads';
export * from './store';
