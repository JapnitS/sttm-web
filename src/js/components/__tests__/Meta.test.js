
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import {
  render,
  screen,
} from '@testing-library/react';
import { toHaveClass } from '@testing-library/jest-dom/matchers'

import { shabad, mockStore } from '../__mocks__/';
import Meta from '../Meta';
expect.extend({ toHaveClass });

jest.mock('react-date-picker', () => () => <div>react calendar</div>)

//NOTE: MemoryRouter - https://testing-library.com/docs/example-react-router#reducing-boilerplate

describe('<Meta />', () => {
  const translationLanguages = ['english', 'spanish', 'punjabi'];
  const transliterationLanguages = ['english'];
  const store = mockStore({ darkMode: false, autoScrollMode: false, showShabadAudioPlayer: true });

  describe('<Meta type=\"ang\" />', () => {
    let renderedComponent = null;

    beforeEach(() => {
      renderedComponent = render(
        <Provider store={store}>
          <Meta
            type="ang"
            isUnicode={false}
            info={shabad.shabadInfo}
            translationLanguages={translationLanguages}
            transliterationLanguages={transliterationLanguages} />
        </Provider >,
        { wrapper: MemoryRouter }
      );
    });

    it('renders correctly', () => {
      const { container } = renderedComponent;
      expect(container).toMatchSnapshot();
    })

    // it('shows navigation arrows properly on fullscreen mode', async () => {
    //   const { container } = renderedComponent;
    //   const fullscreen = container.querySelector('.fullscreen');
    //   await fireEvent.click(fullscreen);

    //   // Basic limitation with css style as we can't test style changes on nav items
    //   const body = document.querySelector('body');
    //   expect(body).toHaveClass('fullscreen-view');
    // });
  });

  describe('<Meta type=\"shabad\" />', () => {
    let renderedComponent = null;
    const store = mockStore({ showShabadAudioPlayer: true });

    beforeEach(() => {
      renderedComponent = render(
        <Provider store={store}>
          <Meta
            type="shabad"
            isUnicode={false}
            info={shabad.shabadInfo}
            translationLanguages={translationLanguages}
            transliterationLanguages={transliterationLanguages} />
        </Provider>,
        { wrapper: MemoryRouter }
      );
    })

    it('renders correctly', () => {
      const { container } = renderedComponent;
      expect(container).toMatchSnapshot();
    })

    it('shows chevron navigation buttons', () => {
      const { queryAllByTestId } = renderedComponent;
      expect(queryAllByTestId('chevron')).toHaveLength(2);
    });
  });

  describe('<Meta type=\"hukamnama\" />', () => {
    let renderedComponent = null;
    beforeEach(() => {
      renderedComponent = render(
        <Provider store={store}>
          <Meta
            type="hukamnama"
            isUnicode={false}
            info={shabad.shabadInfo}
            translationLanguages={translationLanguages}
            transliterationLanguages={transliterationLanguages} />
        </Provider >,
        { wrapper: MemoryRouter }
      );
    })
    it('renders correctly', () => {
      const { container } = renderedComponent;
      expect(container).toMatchSnapshot();
    })
  });
});
