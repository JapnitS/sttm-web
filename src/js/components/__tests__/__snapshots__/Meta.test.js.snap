// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Meta /> <Meta type="ang" /> renders correctly 1`] = `
<div>
  <div
    class="metadata-ang"
    id="metadata"
  >
    <div
      class="shabad-nav left disabled-nav"
    >
      <a>
        <svg
          class="chevron "
          data-testid="chevron"
          height="1em"
          style="transform: rotate(0deg);"
          viewBox="0 0 320 512"
          width="1em"
        >
          <path
            d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
          />
        </svg>
      </a>
    </div>
    <div
      class="meta"
    >
      <h4
        class="gurbani-font"
      >
        rwgu swrMg
         - 
        sRI gurU gRMQ swihb jI
         - 
        <a
          href="/ang?ang=1211&source=G"
        >
          AMg
           
          1211
        </a>
        
      </h4>
      <h4>
        Raag Saarang
         - 
        Sri Guru Granth Sahib Ji
         - 
        <a
          href="/ang?ang=1211&source=G"
        >
          Ang
           
          1211
        </a>
        
      </h4>
      <div
        class="react-audio-player"
      >
        <div
          aria-label="Audio Player"
          class="rhap_container "
          role="group"
          tabindex="0"
        >
          <audio
            preload="auto"
            src="http://old.sgpc.net/hukumnama/jpeg%20hukamnama/hukamnama.mp3"
          />
          <div
            class="rhap_main rhap_stacked"
          >
            <div
              class="rhap_progress-section"
            >
              <div
                class="rhap_time rhap_current-time"
                id="rhap_current-time"
              >
                --:--
              </div>
              <div
                aria-label="Audio Progress Control"
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow="0"
                class="rhap_progress-container"
                role="progressbar"
                tabindex="0"
              >
                <div
                  class="rhap_progress-bar rhap_progress-bar-show-download"
                >
                  <div
                    class="rhap_progress-indicator"
                    style="left: 0%;"
                  />
                  <div
                    class="rhap_progress-filled"
                    style="width: 0%;"
                  />
                </div>
              </div>
              <div
                class="rhap_time rhap_total-time"
              >
                --:--
              </div>
            </div>
            <div
              class="rhap_controls-section"
            >
              <div
                class="rhap_additional-controls"
              />
              <div
                class="rhap_main-controls"
              >
                <button
                  aria-label="Rewind"
                  class="rhap_button-clear rhap_main-controls-button rhap_rewind-button"
                  type="button"
                >
                  <svg
                    focusable="false"
                    height="1em"
                    preserveAspectRatio="xMidYMid meet"
                    style="transform: rotate(360deg);"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M11.5 12l8.5 6V6m-9 12V6l-8.5 6l8.5 6z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
                <button
                  aria-label="Play"
                  class="rhap_button-clear rhap_main-controls-button rhap_play-pause-button"
                  type="button"
                >
                  <svg
                    focusable="false"
                    height="1em"
                    preserveAspectRatio="xMidYMid meet"
                    style="transform: rotate(360deg);"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M10 16.5v-9l6 4.5M12 2A10 10 0 0 0 2 12a10 10 0 0 0 10 10a10 10 0 0 0 10-10A10 10 0 0 0 12 2z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
                <button
                  aria-label="Forward"
                  class="rhap_button-clear rhap_main-controls-button rhap_forward-button"
                  type="button"
                >
                  <svg
                    focusable="false"
                    height="1em"
                    preserveAspectRatio="xMidYMid meet"
                    style="transform: rotate(360deg);"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M13 6v12l8.5-6M4 18l8.5-6L4 6v12z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="rhap_volume-controls"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="shabad-nav right disabled-nav"
    >
      <a>
        <svg
          class="chevron "
          data-testid="chevron"
          height="1em"
          style="transform: rotate(180deg);"
          viewBox="0 0 320 512"
          width="1em"
        >
          <path
            d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;

exports[`<Meta /> <Meta type="hukamnama" /> renders correctly 1`] = `
<div>
  <div
    class="metadata-hukamnama"
    id="metadata"
  >
    <div
      class="meta"
    >
      <div
        class="meta-hukamnama"
      >
        <div
          class="meta-hukamnama-left"
        >
          <div>
            react calendar
          </div>
          <a
            class="hukam-text-link"
          >
            Past Hukamnamas
          </a>
        </div>
        <h4>
          Daily Hukamnama Sri Harmandir Sahib, Amritsar
          , 
          <span />
        </h4>
        <div
          class="meta-hukamnama-right"
        >
          <svg
            style="fill: rgb(1, 102, 155); width: 18px;"
            viewBox="0 0 491.807 491.807"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M485.802 221.184l-184.32-184.32a20.506 20.506 0 00-22.323-4.444c-7.659 3.154-12.636 10.65-12.636 18.924v82.002c-117.842 4.833-220.651 84.398-253.583 198.41C-1.314 381.195-.208 421.356.201 436.491l.082 3.973c0 9.216 6.164 17.306 15.032 19.743 1.823.492 3.645.737 5.448.737a20.51 20.51 0 0017.592-9.994c81.121-136.356 188.6-140.227 228.168-136.376v105.411c0 8.274 4.977 15.77 12.636 18.924 7.68 3.195 16.466 1.413 22.323-4.444l184.32-184.32c8.007-8.01 8.007-20.954 0-28.961z"
            />
          </svg>
          <a
            class="hukamnama-right-link"
            href="/shabad?id=4299"
          >
            Go to shabad
          </a>
        </div>
      </div>
      <h4
        class="gurbani-font"
      >
        rwgu swrMg
         - 
        sRI gurU gRMQ swihb jI
         - 
        <a
          href="/ang?ang=1211&source=G"
        >
          AMg
           
          1211
        </a>
        
      </h4>
      <h4>
        Raag Saarang
         - 
        Sri Guru Granth Sahib Ji
         - 
        <a
          href="/ang?ang=1211&source=G"
        >
          Ang
           
          1211
        </a>
        
      </h4>
      <div
        class="react-audio-player"
      >
        <div
          aria-label="Audio Player"
          class="rhap_container "
          role="group"
          tabindex="0"
        >
          <audio
            preload="auto"
            src="http://old.sgpc.net/hukumnama/jpeg%20hukamnama/hukamnama.mp3"
          />
          <div
            class="rhap_main rhap_stacked"
          >
            <div
              class="rhap_progress-section"
            >
              <div
                class="rhap_time rhap_current-time"
                id="rhap_current-time"
              >
                --:--
              </div>
              <div
                aria-label="Audio Progress Control"
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow="0"
                class="rhap_progress-container"
                role="progressbar"
                tabindex="0"
              >
                <div
                  class="rhap_progress-bar rhap_progress-bar-show-download"
                >
                  <div
                    class="rhap_progress-indicator"
                    style="left: 0%;"
                  />
                  <div
                    class="rhap_progress-filled"
                    style="width: 0%;"
                  />
                </div>
              </div>
              <div
                class="rhap_time rhap_total-time"
              >
                --:--
              </div>
            </div>
            <div
              class="rhap_controls-section"
            >
              <div
                class="rhap_additional-controls"
              />
              <div
                class="rhap_main-controls"
              >
                <button
                  aria-label="Rewind"
                  class="rhap_button-clear rhap_main-controls-button rhap_rewind-button"
                  type="button"
                >
                  <svg
                    focusable="false"
                    height="1em"
                    preserveAspectRatio="xMidYMid meet"
                    style="transform: rotate(360deg);"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M11.5 12l8.5 6V6m-9 12V6l-8.5 6l8.5 6z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
                <button
                  aria-label="Play"
                  class="rhap_button-clear rhap_main-controls-button rhap_play-pause-button"
                  type="button"
                >
                  <svg
                    focusable="false"
                    height="1em"
                    preserveAspectRatio="xMidYMid meet"
                    style="transform: rotate(360deg);"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M10 16.5v-9l6 4.5M12 2A10 10 0 0 0 2 12a10 10 0 0 0 10 10a10 10 0 0 0 10-10A10 10 0 0 0 12 2z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
                <button
                  aria-label="Forward"
                  class="rhap_button-clear rhap_main-controls-button rhap_forward-button"
                  type="button"
                >
                  <svg
                    focusable="false"
                    height="1em"
                    preserveAspectRatio="xMidYMid meet"
                    style="transform: rotate(360deg);"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M13 6v12l8.5-6M4 18l8.5-6L4 6v12z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="rhap_volume-controls"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Meta /> <Meta type="shabad" /> renders correctly 1`] = `
<div>
  <div
    class="metadata-shabad"
    id="metadata"
  >
    <div
      class="shabad-nav left disabled-nav"
    >
      <a>
        <svg
          class="chevron "
          data-testid="chevron"
          height="1em"
          style="transform: rotate(0deg);"
          viewBox="0 0 320 512"
          width="1em"
        >
          <path
            d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
          />
        </svg>
      </a>
    </div>
    <div
      class="meta"
    >
      <h4
        class="gurbani-font"
      >
        rwgu swrMg
         - 
        sRI gurU gRMQ swihb jI
         - 
        <a
          href="/ang?ang=1211&source=G"
        >
          AMg
           
          1211
        </a>
        
      </h4>
      <h4>
        Raag Saarang
         - 
        Sri Guru Granth Sahib Ji
         - 
        <a
          href="/ang?ang=1211&source=G"
        >
          Ang
           
          1211
        </a>
        
      </h4>
      <div
        class="react-audio-player"
      >
        <div
          aria-label="Audio Player"
          class="rhap_container "
          role="group"
          tabindex="0"
        >
          <audio
            preload="auto"
            src="http://old.sgpc.net/hukumnama/jpeg%20hukamnama/hukamnama.mp3"
          />
          <div
            class="rhap_main rhap_stacked"
          >
            <div
              class="rhap_progress-section"
            >
              <div
                class="rhap_time rhap_current-time"
                id="rhap_current-time"
              >
                --:--
              </div>
              <div
                aria-label="Audio Progress Control"
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow="0"
                class="rhap_progress-container"
                role="progressbar"
                tabindex="0"
              >
                <div
                  class="rhap_progress-bar rhap_progress-bar-show-download"
                >
                  <div
                    class="rhap_progress-indicator"
                    style="left: 0%;"
                  />
                  <div
                    class="rhap_progress-filled"
                    style="width: 0%;"
                  />
                </div>
              </div>
              <div
                class="rhap_time rhap_total-time"
              >
                --:--
              </div>
            </div>
            <div
              class="rhap_controls-section"
            >
              <div
                class="rhap_additional-controls"
              />
              <div
                class="rhap_main-controls"
              >
                <button
                  aria-label="Rewind"
                  class="rhap_button-clear rhap_main-controls-button rhap_rewind-button"
                  type="button"
                >
                  <svg
                    focusable="false"
                    height="1em"
                    preserveAspectRatio="xMidYMid meet"
                    style="transform: rotate(360deg);"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M11.5 12l8.5 6V6m-9 12V6l-8.5 6l8.5 6z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
                <button
                  aria-label="Play"
                  class="rhap_button-clear rhap_main-controls-button rhap_play-pause-button"
                  type="button"
                >
                  <svg
                    focusable="false"
                    height="1em"
                    preserveAspectRatio="xMidYMid meet"
                    style="transform: rotate(360deg);"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M10 16.5v-9l6 4.5M12 2A10 10 0 0 0 2 12a10 10 0 0 0 10 10a10 10 0 0 0 10-10A10 10 0 0 0 12 2z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
                <button
                  aria-label="Forward"
                  class="rhap_button-clear rhap_main-controls-button rhap_forward-button"
                  type="button"
                >
                  <svg
                    focusable="false"
                    height="1em"
                    preserveAspectRatio="xMidYMid meet"
                    style="transform: rotate(360deg);"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M13 6v12l8.5-6M4 18l8.5-6L4 6v12z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="rhap_volume-controls"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="shabad-nav right disabled-nav"
    >
      <a>
        <svg
          class="chevron "
          data-testid="chevron"
          height="1em"
          style="transform: rotate(180deg);"
          viewBox="0 0 320 512"
          width="1em"
        >
          <path
            d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
